src/solvers/flattening/boolbv_add_sub.cpp

```diff
@@ -69,17 +76,21 @@ bvt boolbvt::convert_add_sub(const exprt &expr)
 
     const bvt &op=convert_bv(*it);
 
-    if(op.size()!=width)
-      throw "convert_add_sub: unexpected operand width";
+    INVARIANT(
+      op.size() == width,
+      "the width of the bitvector resulting from the expression operand should"
+      "be equal to the width indicated by the expression type");
 
     if(type.id()==ID_vector || type.id()==ID_complex)
     {
       const typet &subtype=ns.follow(type.subtype());
 
       std::size_t sub_width=boolbv_width(subtype);
```

Comment:
While at it: remove the `ns.follow` in here, and let `boolbv_width` deal with possible symbol/tag types (just pass `type.subtype()` to `boolbv_width`.